src/components/tooltip/Tooltip.tsx

```diff
@@ -285,7 +288,7 @@ class Tooltip extends React.Component<TooltipProps, State> {
         }
 
         if (showTooltip) {
-            if (!isLabelMatchingTooltipText || childAriaLabel === undefined) {
+            if (!id && (!isLabelMatchingTooltipText || childAriaLabel === undefined)) {
```

Comment:
it looks like the original issue might be caused because of the wrapping `div` around `Toggle`: https://github.com/box/box-ui-elements/blob/cbe6c855a8c396000624de2d41c3e55643ccfefb/src/features/unified-share-modal/SharedLinkSection.js#L493-L494
it's not adding `aria-describedby` to the correct component

it's possible that removing the wrapping div and moving the appropriate styles from `share-toggle-container` to `toggle-container` would fix the issue